datenguidepy/query_builder.py

```diff
@@ -43,6 +45,7 @@ def __init__(
         parent_field: "Field" = None,
         default_fields: bool = True,
         return_type: str = None,
+        stat_meta_data_provider=None,
```

Comment:
```suggestion
        stat_meta_data_provider: Union ["StatisticsSchemaJsonMetaDataProvider",??] =None,
```
What else than the default could it be? Or is there another way to declare the interface that has to be matched?

Reply:
Thanks for the catch, a typical I'll clean this up later case that was then forgotten. Generally I think there are two ways. Either one explicitly defines and abstract base class and uses this for the type and has the other StatsiticsMetaDataProviders inherit from this class. The other way is not to rely on inheritance, use duck typing and use the mypy user defined protocols to ensure the duck typing. As I prefer duck typing I used the second approach and defined a StatisticsMetaDataProviderProtocol in the query_executioner module (we should probably split this module in the near future). This can then be used in type signatures similar to mypy's `List`,`Dict`,`Iterable` etc.
Both methods (ABC, and Protocols) have the advantange, that if we ever decide to make a change to all StatisticsMetaDataProviders (to include units for example) we get very good feedback on which implementations ave already been updated and which ones haven't been. The same goes for upcoming StatisticsMetaDataProviders, although the only one that might be added soon is the one using the new top level queries.